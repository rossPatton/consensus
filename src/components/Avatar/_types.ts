export type tProps = {
  alt?: string,
  className?: string,
  hash?: string,
  size?: string, // number as string
  type?: 'group' | 'user',
};
