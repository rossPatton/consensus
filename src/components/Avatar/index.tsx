import React, {memo} from 'react';

import {tProps} from './_types';

const prefix = 'https://seccdn.libravatar.org/avatar';

// the component half of the search filter container
// decoupled for ease of placement
const Avatar = memo((props: tProps) => {
  if (typeof props.url !== 'string') return null;

  return (
    <img
      alt={props.alt || 'User or Group Avatar'}
      height={props.size || '96'}
      className="bg-white rounded-circ ovrHide"
      src={`${prefix}/${props.url}?default=${props.type === 'group' ? 'identicon' : 'mm'}&size=${props.size}`}
      width={props.size || '96'}
    />
  );
});

export default Avatar;
