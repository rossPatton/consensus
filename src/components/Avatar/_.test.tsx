import React from 'react';
import render from 'react-test-renderer';

import Avatar from '.';

describe('components/Avatar', () => {
  it('renders without crashing', () => {
    render.create(<Avatar type="groups" />);
  });

  it('renders empty div when no props', () => {
    const component = render.create(<Avatar />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders user image tag when hash prop provided', () => {
    const component = render.create((
      <Avatar
        hash="notARealHash"
        type="users"
      />
    ));
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders group image tag when hash prop provided', () => {
    const component = render.create((
      <Avatar
        hash="notARealHash"
        type="groups"
      />
    ));
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
