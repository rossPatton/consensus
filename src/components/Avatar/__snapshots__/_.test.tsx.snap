// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/Avatar renders empty div when no props 1`] = `
<div
  className="bg-gray-1 rounded-circ overflow-hidden av-sz-lg mr-1"
>
  
</div>
`;

exports[`components/Avatar renders group image tag when hash prop provided 1`] = `
<div
  className="bg-gray-1 rounded-circ overflow-hidden av-sz-lg mr-1"
>
  <img
    alt="User or Group Avatar"
    height="60"
    src="https://consensus.nyc3.digitaloceanspaces.com/groups/notARealHash"
    width="60"
  />
</div>
`;

exports[`components/Avatar renders user image tag when hash prop provided 1`] = `
<div
  className="bg-gray-1 rounded-circ overflow-hidden av-sz-lg mr-1"
>
  <img
    alt="User or Group Avatar"
    height="60"
    src="https://consensus.nyc3.digitaloceanspaces.com/users/notARealHash"
    width="60"
  />
</div>
`;
